microsoft/testsuites/vm_extensions/vmsnapshot_extension.py

```diff
@@ -0,0 +1,106 @@
+)
+class BVTExtension(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        creates a restore point collection then a restore point which helps in
+        validating the VMSnapshot extension.
+        """,
+        priority=1,
+        requirement=simple_requirement(supported_features=[AzureExtension]),
+    )
+    def verify_vmsnapshot_extension(
+        self, log: Logger, node: Node, environment: Environment
+    ) -> None:
+        unique_name = str(uuid.uuid4())
+        node_context = get_node_context(node)
+        resource_group_name = node_context.resource_group_name
+        vm_name = node_context.vm_name
+        node_location = node.capability.get_extended_runbook(AzureNodeSchema, AZURE)
+        location = node_location.location
+        restore_point_collection = "rpc_" + unique_name
+        assert environment.platform
+        platform: AzurePlatform = environment.platform  # type: ignore
+        assert isinstance(platform, AzurePlatform)
+        sub_id = platform.subscription_id
+        # creating restore point collection
+        client = get_compute_client(platform)
+        response = client.restore_point_collections.create_or_update(
+            resource_group_name=resource_group_name,
+            restore_point_collection_name=restore_point_collection,
+            parameters={
+                "location": location,
+                "properties": {
+                    "source": {
+                        "id": "/subscriptions/"
+                        + sub_id
+                        + "/resourceGroups/"
+                        + resource_group_name
+                        + "/providers/Microsoft.Compute/virtualMachines/"
+                        + vm_name
+                    }
+                },
+            },
+        )
+        log.info("restore point collection created")
+        log.info(f"response {response}")
+        count = 0
+
+        while count < 10:
+            try:
+                # create a restore point for the VM
+                restore_point = "rp_" + datetime.now().strftime("%Y-%m-%d-%H-%M-%S")
+                response = client.restore_points.begin_create(
+                    resource_group_name=resource_group_name,
+                    restore_point_collection_name=restore_point_collection,
+                    restore_point_name=restore_point,
+                    parameters={},
+                )
+                wait_operation(response)
+                log.info(f"restore point {restore_point} created")
+                break
+            except Exception as e:
+                # This message is sometimes seen while rp creation
+                # so we will be retrying it after some time
+                if "Changes were made to the Virtual Machine" in str(e):
+                    pass
+                else:
+                    raise e
+            time.sleep(1)
+            count = count + 1
+        assert_that(
+            count, "Restore point creation failed."
```

Comment:
can we format it into single line for better readability.

Reply:
change it into one single line will fix the CI check error